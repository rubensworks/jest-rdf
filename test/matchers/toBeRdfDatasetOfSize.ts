import { defaultGraph, namedNode, quad } from '@rdfjs/data-model';
import datasetFactory = require('rdf-dataset-indexed');
import '../../index';

const quad1 = quad(
  namedNode('s'),
  namedNode('p'),
  namedNode('o'),
  namedNode('g'),
);

const quad2 = quad(
  namedNode('s'),
  namedNode('p'),
  namedNode('o'),
  defaultGraph(),
);

describe('#toBeRdfDatasetOfSize', () => {
  it('should succeed if the dataset has the right number of quad', () => {
    return expect(datasetFactory([quad1, quad2])).toBeRdfDatasetOfSize(2);
  });

  it('should not succeed if the dataset does not have the right number of quads', () => {
    return expect(datasetFactory([quad1])).not.toBeRdfDatasetOfSize(2);
  });

  it('should fail if the dataset does not have the right number of quads', () => {
    return expect(() => expect([
      expect(datasetFactory([quad1])).toBeRdfDatasetOfSize(2),
    ])).toThrowErrorMatchingSnapshot();
  });

  it('should not fail if the dataset has the right number of quads', () => {
    return expect(() => expect([
      expect(datasetFactory([quad1, quad2])).not.toBeRdfDatasetOfSize(2),
    ])).toThrowErrorMatchingSnapshot();
  });
});
