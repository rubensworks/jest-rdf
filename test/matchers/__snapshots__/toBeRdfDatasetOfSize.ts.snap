// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#toBeRdfDatasetOfSize should fail if the dataset does not have the right number of quads 1`] = `"expected dataset to have 2 quads, 1 received"`;

exports[`#toBeRdfDatasetOfSize should not fail if the dataset has the right number of quads 1`] = `"expected dataset to not have exactly 2 quads"`;
